data-prepper-plugins/type-conversion-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/typeconversion/TypeConversionProcessor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.typeconversion;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@DataPrepperPlugin(name = "type_conversion", pluginType = Processor.class, pluginConfigurationType = TypeConversionProcessorConfig.class)
+public class TypeConversionProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private static final Logger LOG = LoggerFactory.getLogger(TypeConversionProcessor.class);
+    private String key;
+    final private String type;
+
+    @DataPrepperPluginConstructor
+    public TypeConversionProcessor(final PluginMetrics pluginMetrics, final TypeConversionProcessorConfig typeConversionProcessorConfig) {
+        super(pluginMetrics);
+        this.key = typeConversionProcessorConfig.getKey();
+        this.type = typeConversionProcessorConfig.getType();
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            Object keyVal = recordEvent.get(key, Object.class);
+            if (keyVal != null) {
+                recordEvent.delete(key);
+                switch (type) {
+                    case "integer" :
+                        if (keyVal instanceof String) {
+                            recordEvent.put(key, Long.parseLong((String)keyVal));
+                        } else if (keyVal instanceof Double) {
+                            recordEvent.put(key, (long)(double)keyVal);
```

Comment:
Why cast twice? `(long)(double)`

Reply:
It looks like java doesn't allow converting Double to Long directly. I was seeing error if I don't cast to `double` first and then to `long`